src/libstd/net/addr.rs

```diff
@@ -554,20 +554,44 @@ impl FromInner<c::sockaddr_in6> for SocketAddrV6 {
 
 #[stable(feature = "ip_from_ip", since = "1.16.0")]
 impl From<SocketAddrV4> for SocketAddr {
+    /// Converts a [`SocketAddrV4`] into a [`V4 variant of SocketAddr`].
+    ///
+    /// [`SocketAddrV4`]: struct.SocketAddrV4.html
+    /// [`V4 variant of SocketAddr`]: enum.SocketAddr.html#variant.V4
```

Comment:
Sorry for my noob question as I haven't touched rust for quite a while.

Is this syntax valid and necessary for the docs?

Reply:
Based on running `cargo doc` on the following, yes I believe it is valid as written.

```rust
pub enum SocketAddr {
    V4,
}

/// Converts a `SocketAddrV4` into a [`V4 variant of SocketAddr`].
///
/// [`V4 variant of SocketAddr`]: enum.SocketAddr.html#variant.V4
pub fn from() {}
```

A more modern way to write it as of https://github.com/rust-lang/rust/issues/43466 would be:

```rust
/// [`V4 variant of SocketAddr`]: SocketAddr::V4
```